miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppSecureStorage.kt

```diff
@@ -0,0 +1,249 @@
+        miniAppId: String,
+        content: String,
+        onSuccess: (Map<String, String>) -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) = try {
+        makeDirectoryAvailable()
+        val fileToWrite = File(secureStorageBasePath, "$miniAppId.txt")
+        if (fileToWrite.exists()) {
+            fileToWrite.delete()
+        }
+        val encryptedFile = EncryptedFile.Builder(
+            activity,
+            fileToWrite,
+            MasterKey.Builder(activity).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build(),
+            EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB
+        ).build()
+
+        val fileContent = content.toByteArray(StandardCharsets.UTF_8)
+        encryptedFile.openFileOutput().apply {
+            write(fileContent)
+            flush()
+            close()
+        }
+        onSuccess(deSerializeItems(content))
+    } catch (e: Exception) {
+        onFailed(MiniAppStorageError.ioError)
+    }
+
+    @VisibleForTesting
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "ReturnCount")
+    private fun readFromEncryptedFile(miniAppId: String): Map<String, String>? {
+        try {
+            if (isStorageAvailable(miniAppId)) {
+                val encryptedFile = EncryptedFile.Builder(
+                    activity,
+                    File(secureStorageBasePath, "$miniAppId.txt"),
+                    MasterKey.Builder(activity).setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+                        .build(),
+                    EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB
+                ).build()
+
+                val inputStream = encryptedFile.openFileInput()
+                val byteArrayOutputStream = ByteArrayOutputStream()
+                var nextByte: Int = inputStream.read()
+                while (nextByte != -1) {
+                    byteArrayOutputStream.write(nextByte)
+                    nextByte = inputStream.read()
+                }
+
+                val plaintext: ByteArray = byteArrayOutputStream.toByteArray()
+                val jsonToRead = plaintext.toString(Charsets.UTF_8)
+                return deSerializeItems(jsonToRead)
+            } else {
+                return null
+            }
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun deleteSecureStorage(
+        miniAppId: String,
+        onSuccess: () -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) {
+        if (isStorageAvailable(miniAppId)) {
+            scope.launch {
+                val file = File(secureStorageBasePath, "$miniAppId.txt")
+                file.delete()
+                if (!file.exists()) {
+                    onSuccess()
+                } else {
+                    onFailed(MiniAppStorageError.failedDeleteError)
+                }
+            }
+        } else {
+            onFailed(MiniAppStorageError.unavailableStorage)
+        }
+    }
+
+    fun deleteSecureStorageItems(
```

Comment:
I think you can simplify the method names in this class to remove `SecureStorage`. i.e. `deleteItems`, etc.